astropy/coordinates/baseframe.py

```diff
@@ -346,7 +346,14 @@ def __init__(
 
         # Broadcast the data if necessary and set it
         if data is not None and data.shape != self._shape:
-            data = data._apply(np.broadcast_to, shape=self._shape, subok=True)
+            try:
+                # if broadcasting isn't strictly needed, avoid it
+                # see https://github.com/astropy/astropy/issues/16219
+                data = data.reshape(self._shape)
```

Comment:
I changed this to a `.reshape` so that a new view is taken - I'm not 100% sure that there aren't some cases in which `data` is just the input, and we should make sure we don't mess that up.

Reply:
good point. Thanks !